daemons/ipa-sam/ipa_sam.c

```diff
@@ -3554,6 +3554,77 @@ static NTSTATUS getsam_interdom_trust_account(struct pdb_methods *methods,
 	return status;
 }
 
+static NTSTATUS ipasam_getsampwsid(struct pdb_methods *methods,
+				    struct samu *user,
+				    const struct dom_sid *sid)
+{
+	struct ipasam_private *ipasam_state =
+			talloc_get_type_abort(methods->private_data, struct ipasam_private);
+	TALLOC_CTX *tmp_ctx;
+	NTSTATUS status;
+	char *filter = NULL;
+	char *sid_str = NULL;
+	LDAPMessage *result = NULL;
+	LDAPMessage *entry = NULL;
+	int ret;
+	int count;
+
+	tmp_ctx = talloc_new(NULL);
+	if (tmp_ctx == NULL) {
+		return NT_STATUS_NO_MEMORY;
+	}
+
+	sid_str = sid_talloc_string(ipasam_state->idmap_ctx, tmp_ctx, sid);
+	if (sid_str == NULL) {
+		return NT_STATUS_NO_MEMORY;
+	}
+
+	filter = talloc_asprintf(tmp_ctx, "(&(|(%s=%s)(%s=%s))(%s=%s))",
+					  LDAP_ATTRIBUTE_OBJECTCLASS,
+					  LDAP_OBJ_SAMBASAMACCOUNT,
+					  LDAP_ATTRIBUTE_OBJECTCLASS,
+					  LDAP_OBJ_ID_OBJECT,
+					  LDAP_ATTRIBUTE_SID, sid_str);
+	if (filter == NULL) {
+		status = NT_STATUS_NO_MEMORY;
+		goto done;
+	}
+
+	ret = smbldap_search(ipasam_state->ldap_state,
+			     ipasam_state->base_dn,
+			     LDAP_SCOPE_SUBTREE,filter, NULL, 0,
+			     &result);
+	if (ret != LDAP_SUCCESS) {
+		status = NT_STATUS_NO_SUCH_USER;
+		goto done;
+	}
+
+	count = ldap_count_entries(priv2ld(ipasam_state), result);
+	if (count != 1) {
+		status = NT_STATUS_NO_SUCH_USER;
```

Comment:
Perhaps log something? This could be difficult to debug.

Reply:
I added, but it is the same for other methods already for quite some time.